examples/step-8/step-8.cc

```diff
@@ -193,12 +193,26 @@ namespace Step8
   // the solution function has, which is <code>dim</code> since we consider
   // displacement in each space direction. The FESystem class can handle this:
   // we pass it the finite element of which we would like to compose the
-  // system of, and how often it shall be repeated:
-
+  // system of, and how often to repeat it. There are different ways to
+  // tell the FESystem constructor how to do this, but the one that is
+  // closest to mathematical notation is to write out what we want to do
+  // mathematically: We want to construct the finite element space
+  // $Q_1^d$ where the index 1 corresponds to the polynomial degree and
+  // the exponent $d$ to the space dimension -- because the *displacement*
+  // we try to simulate here is a vector with exactly $d$ components. The
+  // FESystem class then lets us create this space by initialization with
+  // `FE_Q<dim>(1)^dim`, emulating the mathematical notation.
+  //
+  // (We could also have written `fe(FE_Q<dim>(1), dim)`, which would simply
+  // have called a different constructor of the FESystem class that first
+  // takes the "base element" and then a "multiplicity", i.e., a number that
+  // indicates how many times the base element is to be repeated. The two
+  // ways of writing things are entirely equivalent; we choose the one that
+  // is closer to mathematical notation.)
```

Comment:
Personally, I would switch this. For me the `operator^` is only syntactic sugar (and just an additional layer of indirection). I would show the `operator^` here and keep the *normal* way in all other tutorials. Furthermore, it looks to me that not all compilers are supporting this feature:
https://github.com/dealii/dealii/blob/53084d8c6a5296b727b1d73c5450fe43d4f5d780/include/deal.II/fe/fe_system.h#L499-L502

Reply:
I think the question is what we want the *definitive* way for things to be. My take is to use the `operator^` from here on out. There is not much use in using two different ways to achieve the same thing in the tutorials: I've come to understand that from a teaching perspective, it's useful to pick one way and stick with it, even if there may be other ways to achieve things. We didn't follow this principle in the early years and introduced different ways of doing things just because we could, and I think that's not a good approach and over the years we've tried to unify things on one style.

As for the Intel compiler: What version are they on now? What version do typical supercomputing centers have on their machines?

Reply:
I like the change, and I think that if we can use it everywhere then we should (modernisation of the tutorials is not a bad thing, and the alternative approach that calls the other constructor is mentioned here). I hope that the questions surrounding the compiler incompatibility are not a deal-breaker.